components/datadog/agent/docker.go

```diff
@@ -86,8 +86,8 @@ func dockerAgentComposeManifest(agentImagePath string, apiKey pulumi.StringInput
 						"/sys/kernel/tracing:/sys/kernel/tracing",
 					},
 					Environment: map[string]any{
-						"DD_API_KEY":               apiKeyResolved,
-						"DD_PROCESS_AGENT_ENABLED": true,
+						"DD_API_KEY": apiKeyResolved,
+						"DD_PROCESS_CONFIG_PROCESS_COLLECTION_ENABLED": true,
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
Wonder if we should add a comment as per this variable is compatible with agent 7.35+. I don't expect folks to use the framework with such an old version though.